src/raftstore/store/util.rs

```diff
@@ -326,6 +327,37 @@ pub fn get_region_approximate_size_cf(
     Ok(size)
 }
 
+/// Get the approxmiate middle key of the region. The returned key maybe
+/// is timpstamped if transaction KV is used, and must start with "z".
+pub fn get_region_approximate_middle_cf(
+    db: &DB,
+    cfname: &str,
+    region: &metapb::Region,
+) -> Result<Option<Vec<u8>>> {
+    let cf = rocksdb_util::get_cf_handle(db, cfname)?;
+    let start = keys::enc_start_key(region);
+    let end = keys::enc_end_key(region);
+    let range = Range::new(&start, &end);
+    let collection = db.get_properties_of_tables_in_range(cf, &[range])?;
+
+    let mut keys = Vec::new();
+    for (_, v) in &*collection {
+        let props = SizeProperties::decode(v.user_collected_properties())?;
+        keys.extend(
+            props
+                .index_handles
+                .range::<[u8], _>((Excluded(start.as_slice()), Excluded(end.as_slice())))
```

Comment:
why both are excluded?

Reply:
The middle key is supposed to be used for split regions. So we don't hope the key equals to region's start key or end key.